src/discordtipbot/discordbot.cr

```diff
@@ -122,10 +122,52 @@ class DiscordBot
     end
   end
 
+  private def split(msg : String, spliterator : String, max_length : Int32)
+    msgs = Array(String).new
+
+    # Break up the message into parts which are less than max_length characters each
+    while (msg.size > max_length)
+      # Find the final space before the max_length
+      pos = msg[0, max_length].rindex(spliterator)
+
+      # If there was no space break at the max_length
+      if (pos.nil?)
+        pos = max_length
+      end
+
+      # Record that message!
+      msgs << msg[0, pos]
+
+      # Find the index of the next space after the pos
+      i = msg.index(spliterator, pos)
+
+      if (i.nil?)
+        i = 0
+      else
+        i += 1
+      end
+
+      # If the next space is more than max_length away just record it as the break pos.
+      # FIXME: This will cause problems if a word is longer than max_length! (it is unexpected that
+      #         this will happen with the tipbot however)
+      if (i < pos || i > pos + max_length)
+        i = pos
+      end
+
+      # Split the message at this next space to prepare it for the following iteration
+      msg = msg[i, msg.size - i]
+    end
+    msgs << msg
+
+    return msgs
+  end
+
   # Since there is no easy way, just to reply to a message
   private def reply(payload : Discord::Message, msg : String)
+    msgs = split(msg, " ", 2000)
```

Comment:
There's no reason to do this for every `reply`.

Remove this so we can still implicitly `begin` and:
```cr
if msg.size > 2000
  # split send
else
  # create single message
end
````